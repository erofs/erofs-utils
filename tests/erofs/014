#!/bin/sh
# SPDX-License-Identifier: GPL-2.0+
#
# check if cross-device submounts are handled properly
#
seq=`basename $0`
seqres=$RESULT_DIR/$(echo $0 | awk '{print $((NF-1))"/"$NF}' FS="/")

# get standard environment, filters and checks
. "${srcdir}/common/rc"

cleanup()
{
	cd /
	rm -rf $tmp.*
}

_require_root
_require_erofs

# remove previous $seqres.full before test
rm -f $seqres.full

# real QA test starts here
echo "QA output created by $seq"

if [ -z $SCRATCH_DEV ]; then
	SCRATCH_DEV=$tmp/erofs_$seq.img
	rm -f SCRATCH_DEV
fi

localdir="$tmp/$seq"
rm -rf $localdir

tds="$localdir/s"
tmg1="$localdir/s1.img"
tmg2="$localdir/s2.img"
tdd1="$localdir/d/d1"
tdd2="$localdir/d/d2"

mkdir -p $tds
touch $tds/t
$MKFS_EROFS_PROG $MKFS_OPTIONS $tmg1 $tds >> $seqres.full 2>&1
cp $tmg1 $tmg2

mkdir -p $tdd1 $tdd2
$MOUNT_PROG -t erofs -o loop $tmg1 $tdd1
$MOUNT_PROG -t erofs -o loop $tmg2 $tdd2
_scratch_mkfs $localdir/d >> $seqres.full 2>&1 || _fail "failed to mkfs"

$UMOUNT_PROG $tdd1
$UMOUNT_PROG $tdd2

_scratch_mount 2>>$seqres.full

[ "x`stat -c %i $SCRATCH_MNT/d1/t`" = "x`stat -c %i $SCRATCH_MNT/d2/t`" ] && \
	_fail "-> buggy cross-device sub-mount"

_scratch_unmount

echo Silence is golden
status=0
exit 0
