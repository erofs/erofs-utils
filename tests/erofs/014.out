QA output created by 014
Silence is golden
